ci/nix.yml

```diff
@@ -0,0 +1,20 @@
+name: Nix
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cachix/install-nix-action@v10
+
+    - name: Build Flakes
+      run: nix-build -E '(import (fetchTarball https://github.com/edolstra/flake-compat/archive/master.tar.gz) { src = ./.; }).defaultNix.packages.${builtins.currentSystem}'
```

Comment:
Flakes are still experimental technology that can change any time so I would be wary of promoting it to people who did not opt in into that.